### PR TITLE
Authentication

### LINKED ISSUE #61: Feature: Authentication
**Todo**
- [x] User can login (w/ wallet)
- [x] User  can switch network
- [x] User can logout
- [x] integration of the auth component in the header
- [x] automatic reconnection

image

https://www.figma.com/file/v53wiSwcL5DYynVL0z6L4s/Fuji---Working-File-(NEW)?node-id=156%3A3847

### PR DESCRIPTION
**Pull request by doliG**
close #61 

To fix:
- [x] when an unsupported chain is selected, app crashes on load
- [x] when already in the app and select an unsupported chain, no selected value in the dropdown
- [x] when I switch the address from my wallet, it doesn't change the address in the auth bar
- [x] overlapping elements
- [x] When I visit for the first time and I don't have any addresses connected, MM prompts me to connect. The prompt should display only when the user clicks on "Connect Wallet".
- [x] This is not directly related to this PR but when I try to run a production build with yarn build, there is a bunch of errors (lining problems, implicit any and more).

### CODE REVIEW

**Review by brozorec — CHANGES_REQUESTED** [review]
Some issues I spotted:
- when an unsupported chain is selected, app crashes on load
- when already in the app and select an unsupported chain, no selected value in the dropdown
- when I switch the address from my wallet, it doesn't change the address in the auth bar
- overlapping elements 
image

**Comment by brozorec** [comment]
1. The issues [here](https://github.com/Fujicracy/fuji-v2/pull/72#issue-1399800352) are fixed :+1: 

2. However, I found another one. When I visit for the first time and I don't have any addresses connected, MM prompts me to connect. The prompt should display only when the user clicks on "Connect Wallet".

3. This is not directly related to this PR but when I try to run a production build with `yarn build`, there is a bunch of errors (lining problems, implicit `any` and more).

@doliG

File: packages/frontend/components/Header.tsx [hunk 1]
```diff
@@ -145,15 +153,28 @@ const Header = () => {
...
+                      // TODO: imoprove store typing
+                      balance={balance}
+                      address={address as string}
+                    />
+                  </Grid>
+                  <Grid item>
+                    <ParametersModal />
+                  </Grid>
+                </>
+              )}
```
Comment (Chris-CHEVALIER) [inline packages/frontend/components/Header.tsx:188]: Why not using a ternary operator to handle these conditions?

File: packages/frontend/store/index.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+type Action = {
+  login: () => void
+}
+
+export const useStore = create<State & Action>((set) => ({
+  status: "disconnected",
+  address: "",
+  balance: "",
```
Comment (Chris-CHEVALIER) [inline packages/frontend/store/index.ts]: Balance state is a string, a number or an array?

File: packages/frontend/store/index.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+  login: async () => {
+    const wallets = await onboard.connectWallet()
+    if (!wallets[0]) {
+      set({
+        status: "disconnected",
+      })
+      throw "Cannot login"
+    }
+    console.log(wallets[0])
```
Comment (Chris-CHEVALIER) [inline packages/frontend/store/index.ts]: Don't forget to remove "console.log()" before PR ;)

File: packages/frontend/store/index.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+export const useStore = create<State & Action>((set) => ({
+  status: "disconnected",
+  address: "",
+  balance: "",
+
+  login: async () => {
+    const wallets = await onboard.connectWallet()
+    if (!wallets[0]) {
```
Comment (Chris-CHEVALIER) [inline packages/frontend/store/index.ts]: Why don't use a try / catch with exception handling?

**Pull request merged**